src/Warning/WarningFactory.php

```diff
@@ -12,14 +12,20 @@
 final class WarningFactory
 {
     private WarningFormatter $warningFormatter;
+    private array $excludedStatementsAsString = [];
 
     /**
-     * @param StatementInterface[] $statements
-     * @param string[]             $criticalTables
+     * @param StatementInterface[]      $statements
+     * @param string[]                  $criticalTables
+     * @param StatementInterface[]|null $excludedStatements
      */
-    public function __construct(private readonly iterable $statements, private readonly array $criticalTables = [])
+    public function __construct(private readonly iterable $statements, private readonly array $criticalTables = [], array $excludedStatements = [])
     {
         $this->warningFormatter = new WarningFormatter();
+
+        foreach ($excludedStatements as $excludedStatement) {
```

Comment:
should be done in the extension cf https://github.com/ismail1432/safe-migrations-bundle/pull/5/files#r1190087996